File: probfilter-core/src/main/java/com/c6h5no2/probfilter/hash/Funnel.java
package com.c6h5no2.probfilter.hash;

import java.io.Serializable;


/**
 * A funnel that unpacks an object as primitive values.
 *
 * @see Funnels$ Funnels
 */
@FunctionalInterface
public interface Funnel<T> extends Serializable {
    /**
     * Unpacks {@code T from} as primitives into {@link Sink} {@code into}.
     */
    void apply(T from, Sink into);
}

File: probfilter-core/src/main/scala/com/c6h5no2/probfilter/hash/Funnel.scala
package com.c6h5no2.probfilter.hash


/**
 * A funnel that unpacks an object as primitive values.
 *
 * @see Funnels
 */
trait Funnel[T] extends Serializable {
  /**
   * Unpacks `T` `from` as primitives into `Sink` `into`.
   */
  def apply(from: T, into: Sink): Unit
}
